website/docs/r/dedicated_hardware_security_module.html.markdown

```diff
@@ -0,0 +1,158 @@
+  address_prefixes     = ["10.2.1.0/24"]
+
+  delegation {
+    name = "first"
+
+    service_delegation {
+      name = "Microsoft.HardwareSecurityModules/dedicatedHSMs"
+      actions = [
+        "Microsoft.Network/networkinterfaces/*",
+        "Microsoft.Network/virtualNetworks/subnets/join/action",
+      ]
+    }
+  }
+}
+
+resource "azurerm_subnet" "example3" {
+  name                 = "gatewaysubnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.2.255.0/26"]
+}
+
+resource "azurerm_public_ip" "example" {
+  name                = "example-pip"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  allocation_method   = "Dynamic"
+}
+
+resource "azurerm_virtual_network_gateway" "example" {
+  name                = "example-vnetgateway"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  type     = "ExpressRoute"
+  vpn_type = "PolicyBased"
+  sku      = "Standard"
+
+  ip_configuration {
+    public_ip_address_id          = azurerm_public_ip.example.id
+    private_ip_address_allocation = "Dynamic"
+    subnet_id                     = azurerm_subnet.example3.id
+  }
+}
+
+resource "azurerm_dedicated_hardware_security_module" "example" {
+  name                = "example-hsm"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku_name            = "SafeNet Luna Network HSM A790"
+
+  network_profile {
+    network_interface_private_ip_addresses = ["10.2.1.8"]
+    subnet_id                              = azurerm_subnet.example2.id
+  }
+
+  stamp_id = "stamp2"
+
+  tags = {
+    env = "Test"
+  }
+
+  depends_on = [azurerm_virtual_network_gateway.example]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Dedicated Hardware Security Module. Changing this forces a new Dedicated Hardware Security Module to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Dedicated Hardware Security Module should exist. Changing this forces a new Dedicated Hardware Security Module to be created.
+
+* `location` - (Required) The Azure Region where the Dedicated Hardware Security Module should exist. Changing this forces a new Dedicated Hardware Security Module to be created.
+
+* `sku_name` - (Required) The sku name of the dedicated hardware security module. Changing this forces a new Dedicated Hardware Security Module to be created.
+
+* `network_profile` - (Required)  A `network_profile` block as defined below.
+
+* `stamp_id` - (Required) The ID of the stamp. Changing this forces a new Dedicated Hardware Security Module to be created.
```

Comment:
Could we order these alphabetically o asin schema? 

as well as list the possible values for stand id given there are only two allowed?

Reply:
updated